webapi/src/main/scala/org/knora/webapi/messages/StringFormatter.scala

```diff
@@ -2337,16 +2392,18 @@ class StringFormatter private (
    * @param errorFun   a function that throws an exception. It will be called if the string cannot be parsed.
    * @return the decoded UUID.
    */
-  def validateBase64EncodedUuid(base64Uuid: String, errorFun: => Nothing): UUID = { // V2 / value objects
-    val decodeTry = Try {
-      base64DecodeUuid(base64Uuid)
-    }
+  @deprecated("Use validateBase64EncodedUuid(String) instead.")
+  def validateBase64EncodedUuid(base64Uuid: String, errorFun: => Nothing): UUID = // V2 / value objects
+    validateBase64EncodedUuid(base64Uuid).getOrElse(errorFun)
 
-    decodeTry match {
-      case Success(uuid) => uuid
-      case Failure(_)    => errorFun
-    }
-  }
+  def validateBase64EncodedUuid(base64Uuid: String): Validation[Throwable, UUID] =
+    Validation.fromTry(
+      Try(base64DecodeUuid(base64Uuid))
+        .fold(
+          _ => Failure(ValidationException(s"String: '$base64Uuid' is not a valid base 64 UUID.")),
+          uuid => Success(uuid)
+        )
```

Comment:
does it need the `uuid => Success(uuid)`? I think that's just identity, right?  
If so, you could do something like `orElseFailWith` or `mapError`

Reply:
fixed